cdislogging/__init__.py

```diff
@@ -62,32 +69,39 @@ def get_logger(logger_name, file_name=None, log_level='debug'):
     """
 
     log_levels = {                  # sorted level
+        'notset': logging.NOTSET,   # 00
         'debug': logging.DEBUG,     # 10
         'info': logging.INFO,       # 20
         'warning': logging.WARNING, # 30
         'warn': logging.WARNING,    # 30
         'error': logging.ERROR,     # 40
     }
 
-    if log_level not in log_levels:
-        error_message = 'Invalid log_level parameter: {}\n\n' \
-                        'Valid options: debug, info, warning, ' \
-                        'warn, error'.format(log_level)
-        raise Exception(error_message)
-
     logger = logging.getLogger(logger_name)
 
-    # If at least one log handler exists that means it has been
-    # instantiated with the same name before. Do not keep creating handlers
-    # or your logs will be very messy.
-    if logger.handlers:
-        return logger
+    if log_level:
+        if log_level not in log_levels:
```

Comment:
if you wanna be fancy this can be one line `if log_level and log_level not in log_levels:`

Reply:
Tru. So then L89 has to turn into an elif? 
```
if log_level and log_level not in log_levels: 
    [raise exception]
elif log_level:
    [set level]
```
This block just doesn't read well because the Exception part is big and then the set level part is a oneliner after a line break: 
```
if log_level and log_level not in log_levels: 
    [raise exception]
    [is a longish]
    [block of code, vertically]
    [reader has forgotten about condition]
elif log_level:
    [set level]
```
So maybe I can do instead:
```
if log_level and log_level in log_levels:
    [set level]
elif log_level:
    [raise exception]
    [blah]
    [blah]
```
and I think that will be less backward and read better? 
I confess I prefer the current "catch error if needed else continue" pattern to "if a and b (thing) elif just a (raise error)" but yeah it ends up pretty derpy just layout-wise

Reply:
oh shoot, I didn't think this through or actually consider what was in the block. I also generally prefer the "catch error if needed and handle, continue" pattern to reduce the nesting where possible. but I'm sometimes privvy to the third option you provided too so the actual thing that should happen is first, so I guess I'm inconsistent :man_facepalming: you can effectively ignore me, any way is fine, not a huge deal